ansible/roles/version/tasks/main.yaml

```diff
@@ -16,11 +16,11 @@
 #                the build tag.
 - name: determine kubernetes images
   shell: |
-    kubeadm config images list
-      --image-repository '{{ k8s_image_registry }}' |
-    sed 's/^\(.\+etcd:.\+\)-\([0-9]\+\)$/\1{{ "-\2" if not kubernetes_build_tag }}/' |
+    kubeadm config images list --kubernetes-version='{{ kubernetes_semver }}' --image-repository '{{ k8s_image_registry }}' |
+    sed 's/^\(.\+etcd:\)\(.\+\)-\([0-9]\+\)$/\1{{ "\2-\3" if not kubernetes_build_tag else "v\2"}}/' |
     sed 's/$/{{ '_' + kubernetes_build_tag if kubernetes_build_tag }}/' |
-    sed 's/^\(.\+pause:.\+\)_{{ kubernetes_build_tag }}$/\1/'
+    sed 's/^\(.\+pause:.\+\)_{{ kubernetes_build_tag }}$/\1/' |
+    sed 's#^\(.\+coredns:\)\(.\+\)_{{ kubernetes_build_tag }}#{{ coredns_image_registry_repository }}:\2#'
```

Comment:
a bit ugly fix to make `etcd` and `coredns` images work for both FIPS and non-FIPS versions. 
I intend to rewrite it in future.

Reply:
Non FIPS images:
```
(item=k8s.gcr.io/kube-apiserver:v1.21.6)
(item=k8s.gcr.io/kube-controller-manager:v1.21.6)
(item=k8s.gcr.io/kube-scheduler:v1.21.6)
(item=k8s.gcr.io/kube-proxy:v1.21.6)
(item=k8s.gcr.io/pause:3.4.1)
(item=k8s.gcr.io/etcd:3.4.13-0)
(item=k8s.gcr.io/coredns/coredns:v1.8.0)
(item=k8s.gcr.io/pause:3.2)
(item=docker.io/mesosphere/kfips:v0.1.0)
```

FIPS changes:
```
(item=docker.io/mesosphere/kube-apiserver:v1.21.6_fips.0)
(item=docker.io/mesosphere/kube-controller-manager:v1.21.6_fips.0)
(item=docker.io/mesosphere/kube-scheduler:v1.21.6_fips.0)
(item=docker.io/mesosphere/kube-proxy:v1.21.6_fips.0)
(item=docker.io/mesosphere/pause:3.4.1)
(item=docker.io/mesosphere/etcd:v3.4.13_fips.0)
(item=k8s.gcr.io/coredns/coredns:v1.8.0)
(item=k8s.gcr.io/pause:3.2)
(item=docker.io/mesosphere/kfips:v0.1.0)
```